testing/board/test_board_dc.cpp

```diff
@@ -0,0 +1,319 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+extern "C" {
+
+#include <json-c/json.h>
+#include <uuid/uuid.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include "opae_int.h"
+#include "safe_string/safe_string.h"
+}
+
+#include <opae/fpga.h>
+#include "intel-fpga.h"
+#include <linux/ioctl.h>
+
+#include <cstdlib>
+#include <string>
+#include <netinet/ether.h>
+#include "gtest/gtest.h"
+#include "test_system.h"
+#include "libboard/board_dc/board_dc.h"
+
+
+using namespace opae::testing;
+
+class board_dc_c_p : public ::testing::TestWithParam<std::string> {
+protected:
+	board_dc_c_p() : tokens_{ {nullptr, nullptr} } {}
+
+	fpga_result write_sysfs_file(const char *file,
+		void *buf, size_t count);
+	ssize_t eintr_write(int fd, void *buf, size_t count);
+	fpga_result delete_sysfs_file(const char *file);
+
+	virtual void SetUp() override {
+		ASSERT_TRUE(test_platform::exists(GetParam()));
+		platform_ = test_platform::get(GetParam());
+		system_ = test_system::instance();
+		system_->initialize();
+		system_->prepare_syfs(platform_);
+
+		filter_ = nullptr;
+		ASSERT_EQ(fpgaInitialize(NULL), FPGA_OK);
+		ASSERT_EQ(fpgaGetProperties(nullptr, &filter_), FPGA_OK);
+		ASSERT_EQ(fpgaPropertiesSetObjectType(filter_, FPGA_DEVICE), FPGA_OK);
+		num_matches_ = 0;
+		ASSERT_EQ(fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(),
+			&num_matches_), FPGA_OK);
+		EXPECT_GT(num_matches_, 0);
+		dev_ = nullptr;
+		ASSERT_EQ(fpgaOpen(tokens_[0], &dev_, 0), FPGA_OK);
+	}
+
+	virtual void TearDown() override {
+		EXPECT_EQ(fpgaDestroyProperties(&filter_), FPGA_OK);
+		if (dev_) {
+			EXPECT_EQ(fpgaClose(dev_), FPGA_OK);
+			dev_ = nullptr;
+		}
+		for (auto &t : tokens_) {
+			if (t) {
+				EXPECT_EQ(fpgaDestroyToken(&t), FPGA_OK);
+				t = nullptr;
+			}
+		}
+		fpgaFinalize();
+		system_->finalize();
+	}
+
+	std::array<fpga_token, 2> tokens_;
+	fpga_properties filter_;
+	fpga_handle dev_;
+	test_platform platform_;
+	uint32_t num_matches_;
+	test_system *system_;
+};
+
+ssize_t board_dc_c_p::eintr_write(int fd, void *buf, size_t count)
+{
+	ssize_t bytes_written = 0, total_written = 0;
+	char *ptr = (char*)buf;
+
+	if (!buf) {
+		return -1;
+	}
+
+	while (total_written < (ssize_t)count) {
+		bytes_written =
+			write(fd, ptr + total_written, count - total_written);
+		if (bytes_written < 0) {
+			if (errno == EINTR) {
+				continue;
+			}
+			return bytes_written;
+		}
+		total_written += bytes_written;
+	}
+	return total_written;
+
+}
+fpga_result board_dc_c_p::write_sysfs_file(const char *file,
+	void *buf, size_t count) {
+	fpga_result res = FPGA_OK;
+	char sysfspath[FPGA_VAR_BUF_LEN];
+	int fd = 0;
+
+	snprintf_s_ss(sysfspath, sizeof(sysfspath), "%s/%s", "/sys/class/fpga/intel-fpga-dev.0/intel-fpga-fme.0", file);
+
+	glob_t pglob;
+	int gres = glob(sysfspath, GLOB_NOSORT, NULL, &pglob);
+	if ((gres) || (1 != pglob.gl_pathc)) {
```

Comment:
Shouldn't this check pglob.gl_pathv for NULL?

Reply:
fixed